src/optimizer/__tests__/optimizer-integration-test.js

```diff
@@ -35,6 +35,32 @@ describe('optimizer-integration-test', () => {
     optimized = /[\da-z-]/;
 
     expect(optimizer.optimize(original).toString()).toBe(optimized.toString());
+
+    original = /[-0^]/;
+    optimized = /[0^-]/;
+
+    expect(optimizer.optimize(original).toString()).toBe(optimized.toString());
+
+    original = /[0-^]/;
+    optimized = /[0-^]/;
+
+    expect(optimizer.optimize(original).toString()).toBe(optimized.toString());
+
+    original = /[0^-]/;
+    optimized = /[0^-]/;
+
+    expect(optimizer.optimize(original).toString()).toBe(optimized.toString());
+
+    original = /[^-]/;
+    optimized = /[^-]/;
+
+    expect(optimizer.optimize(original).toString()).toBe(optimized.toString());
+
+    // TODO: Fix this
+    original = /[-^]/;
+    optimized = /[^-]/;
```

Comment:
This still broken, I'm not sure how to fix it. Maybe add some unsortable case?

Reply:
Yeah, this seems a regression, current version doesn't produce `/[^-]/` in this case. I think we should treat `^` as unsortable generically.

Reply:
Actually, maybe returning `return Number.MAX_SAFE_INTEGER` both for `-` and `^` would help?

Reply:
No, `[-^]` `[^-]` has to be different, but current solution will break one of them, whatever we choose the order it. We have to find a different solution.

Reply:
I see – yes, then maybe putting them at the very end instead or adding extra slashes post-sorting?